### PR TITLE
Allow user to switch between his/her organizations

### PR DESCRIPTION
**Pull request by awel1122**
# Allow user to switch between his/her organizations

## changes
    - allows user to switch between organizations it belongs
    - changes user permissions according to organization in use
    - sets default organization to selected active organization

### [Trello card #382](https://trello.com/c/5Wux5OQc/382-frontend-as-user-of-multiple-organizations-i-want-to-change-organization-and-permissions) [#358](https://trello.com/c/GXaGi9Aa/358-bugi-k%C3%A4ytt%C3%A4j%C3%A4-ei-voi-vaihtaa-organisaatiota-tapahtumaa-lis%C3%A4tess%C3%A4-jos-k%C3%A4ytt%C3%A4j%C3%A4tasot-ovat-organisaatioissa-eri)

-----------------------------------------------------------------------------------------------
### Breakdown:

 1. src/actions/tests/userActions.test.js
     * adds organization switching related tests to userActions
   
 2. src/actions/user.js
     * adds setting user active organization action and updates permissions
    
 3. src/components/Header/ActiveOrganizationSelector.js
     * adds ActiveOrganizationSelector component
     
 4. src/components/Header/ActiveOrganizationSelector.test.js 
     * adds tests for ActiveOrganizationSelector component
   
 5. src/components/Header/Header.test.js
     * adds tests to Header component
    
 6. src/components/Header/index.js
      * utilizes ActiveOrganizationSelector in Header
      
 7. src/components/Header/index.scss
      * adds styling related to ActiveOrganizationSelector
      
 8. src/reducers/tests/userReducer.test.js
      * adds test related to setting active organization
      
 9. src/reducers/user.js
      * sets user active organization to state
      
 10. src/constants.js
      * adds `SET_ACTIVE_ORGANIZATION `to constants
      
 11. src/components/FormFields/index.js
      * sets default organization to active organization
       
 12. src/components/FormFields/FormFields.test.js
      * adds test for setting activeOrganization as default organization
      
 13. src/views/Editor/index.js
      * passes activeOrganization prop to FormFields

### CODE REVIEW

**Review by hienous — CHANGES_REQUESTED** [review]
observations/findings should be viewable in this gist. https://gist.github.com/hienous/4b75312bc70f2f4bbc97d44049942ad3

**Comment by awel1122** [comment]
> observations/findings should be viewable in this gist. https://gist.github.com/hienous/4b75312bc70f2f4bbc97d44049942ad3
fetch_all_organizations

Fixed as follows. [Backend](https://github.com/City-of-Turku/linkedevents/pull/131) change contains pagination-class which will send 1000 pages (if available) and with that we don't need to changed organization fetching logic in frontend.

File: src/components/Header/ActiveOrganizationSelector.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            const userOrganizations = organizations.filter(org => userOrganizationIds.includes(org.id))
+            this.setState(() => ({userOrganizations: userOrganizations}))
+            if (!this.state.activeOrganization && !!activeOrganization) {
+                const defaultOrg = userOrganizations.find((org)=> org.id === activeOrganization)
+                this.setState(() => ({activeOrganization: defaultOrg}))
+            }
+        }
+        document.addEventListener('click', this.handleClick, false);
+    }
+
```
Comment (hienous) [inline src/components/Header/ActiveOrganizationSelector.js:29]: If the user only belongs to one organization -> the actual html isn't being rendered but the event listener is still set which causes errors when clicking anywhere on the site. Maybe the event listener should only be added if there are more than 1 organizations.
image

**Review by hienous — CHANGES_REQUESTED** [review]

File: src/components/Header/ActiveOrganizationSelector.js (hunk 1 above)
Comment (awel1122) [reply src/components/Header/ActiveOrganizationSelector.js:29]: Fixed.

**Review by hienous — APPROVED** [review]

**Pull request merged**